### ISSUE TITLE
`@JacksonXmlRootElement` malfunction when using it with multiple `XmlMapper`s and disabling annotations

### ISSUE THREAD

**Issue by benej60** [FasterXML/jackson-dataformat-xml#282]
Found this in version 2.9.4 running some tests that go back and forth serializing with an XML mapper that uses annotations, and another one that ignores them. May be related to issue #171 and the cache of class annotations.

When running this code, the second print statement should use the annotation's localName but it instead uses the class name.

```
@JacksonXmlRootElement(localName = "myname")
public class XMLTest {

    public static void main(String[] s) throws Exception {

        final ObjectMapper xmlMapper = new XmlMapper();
        final ObjectMapper noAnnotationsXmlMapper = xmlMapper.copy()
                .configure(MapperFeature.USE_ANNOTATIONS, false)
                .configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false);

        System.out.println(noAnnotationsXmlMapper.writeValueAsString(new XMLTest()));
        System.out.println(xmlMapper.writeValueAsString(new XMLTest()));

    }

}
```

Output:

```
<XMLTest/>
<XMLTest/>
```

**Comment by cowtowncoder** [comment]
Ok, does sound like caching of root names might be causing the issue here.

**Comment by cowtowncoder** [comment]
Turned out to be inadvertent sharing of root name lookup cache. Was resolved for 3.0 (due to the way construction changed), but test now covers 2.9 and 3.0. Fix will be in `2.9.6`.
